pandas-stubs/core/frame.pyi

```diff
@@ -157,6 +157,7 @@ class _LocIndexerFrame(_LocIndexer):
         self,
         idx: IndexType
         | MaskType
+        | Callable[[DataFrame], IndexType | MaskType | list[HashableT]]
```

Comment:
Does pandas support callables for both index and columns? Might need to also add it to the tuple format.

Reply:
> Does pandas support callables for both index and columns? Might need to also add it to the tuple format.

Yes.  Question is whether I try to get this right now (and create some tests), or wait for a bug report.  If you think I should add it now, let me know and I'll give it a shot.

Reply:
I'm fine adding annotations as needed :) Only popped out because in one place a plain Callable is used and in the other place the argument/return type is specified.

Reply:
Happy to merge after the conflict with main is resolved

Reply:
should be done in https://github.com/pandas-dev/pandas-stubs/pull/509/commits/fdf4e0340d82d93d73eafbcd18d92eca7488ea0b

Reply:
> I'm fine adding annotations as needed :) Only popped out because in one place a plain Callable is used and in the other place the argument/return type is specified.

Yes, I saw that and decided not to try to fix it.  The other issue is that we match on a `Hashable` and any `Callable` works there too.